runtime/utils/common/src/aws/smithy/kotlin/runtime/util/Attributes.kt

```diff
@@ -123,4 +123,7 @@ private class AttributesImpl : Attributes {
 
     override val keys: Set<AttributeKey<*>>
         get() = map.keys
+
+    override fun equals(other: Any?): Boolean =
+        other is AttributesImpl && map == other.map
```

Comment:
**Question**: Same question above about checking specific implementations vs the interface.